packages/plugin-vscode/src/server.ts

```
@@ -32,17 +35,76 @@ type GroupedCompletionItemsType = {
 
 const tokenGroupPatterns: TokenGroupPatternsType = {
   color:
-    /color|background|shadow|border[^-radius]|column-rule|filter|opacity|outline|text-decoration/,
-  space: /margin|padding|gap|top|left|right|bottom/,
+    /color|background|shadow|border[^-radius]|column-rule|filter|opacity|outline|text-decoration|caret-color|accent-color|border-(top|right|bottom|left)-color|background-color|border-color|outline-color|text-decoration-color|column-rule-color/,
+  space:
+    /margin|padding|gap|top|left|right|bottom|inset|margin-(top|right|bottom|left|block|inline)|margin-(block|inline)-(start|end)|padding-(top|right|bottom|left|block|inline)|padding-(block|inline)-(start|end)|row-gap|column-gap|inset-(block|inline)|inset-(block|inline)-(start|end)/,
   'font-size': /font-size|font(?!-)/,
   'font-weight': /font-weight|font(?!-)/,
   'line-height': /line-height|font(?!-)/,
   'letter-spacing': /letter-spacing|font(?!-)/,
   font: /font-family|font(?!-)/,
-  radius: /border-radius/,
-  shadow: /shadow|filter/,
-  blur: /blur|filter/
+  radius:
+    /border-radius|border-(top|bottom)-(left|right)-radius|border-(start|end)-(start|end)-radius/,
+  shadow: /shadow|box-shadow|text-shadow|drop-shadow|filter/,
+  blur: /blur|backdrop-filter|filter/
 };
+const settings = { maxNumberOfProblems: 1000 };
+
+async function validateDocument(
+  textDocument: TextDocument
+): Promise<Diagnostic[]> {
+  const text = textDocument.getText();
```

Considering handling the errors in validateDocument and other places.

---

This function will always get called with a valid document, and the getText() function will return a string, so try catch is not needed here.  

Apart from this VSCode api is flexible and ignores error and functions.